### PR TITLE
Add FX table schema

### PR DESCRIPTION
**Pull request by mi6091**
* Adds reproducible FX database schema
* Adds FX csv import code

### CODE REVIEW

File: enterprai-liquibase/dtcc-db-changelog/dtcc-changelog-1.0.xml [hunk 1]
```diff
@@ -77,4 +77,75 @@
...
 
-</databaseChangeLog>
\ No newline at end of file
+    <changeSet id="2" author="mihailkovachev">
+        <createTable tableName="dtcc_fx">
+
+            <column name="dissemination_id" type="VARCHAR(32)">
+                <constraints nullable="false" primaryKey="true" primaryKeyName="pk_dtcc_fx_dissemination_id"/>
+            </column>
+            <column name="original_dissemination_id" type="VARCHAR(32)" />
```
Comment (du210) [inline enterprai-liquibase/dtcc-db-changelog/dtcc-changelog-1.0.xml:86]: Same as `dissemination_id`, could be of type nullable `int`.

File: enterprai-liquibase/dtcc-db-changelog/dtcc-changelog-1.0.xml [hunk 2]
```diff
@@ -77,4 +77,75 @@
...
+            <column name="end_date" type="date" />
+
+            <!-- <column name="day_count_convention" type="varchar(32)" /> -->
+
+            <column name="settlement_currency" type="VARCHAR(3)" />
+            <column name="asset_class" type="VARCHAR(64)" />
+            <!--<column name="sub_asset_class_for_other_commodity" type="VARCHAR(64)" /> -->
+            <column name="taxonomy" type="VARCHAR(64)" />
+            <column name="price_forming_continuation_data" type="VARCHAR(32)" />
+            <column name="underlying_asset_1" type="VARCHAR(128)" /> <!-- date?? -->
```
Comment (du210) [inline enterprai-liquibase/dtcc-db-changelog/dtcc-changelog-1.0.xml:115]: Yea, don't know why. Let's just ignore for now.

File: enterprai-liquibase/dtcc-db-changelog/dtcc-changelog-1.0.xml [hunk 3]
```diff
@@ -77,4 +77,75 @@
...
+            </column>
+            <column name="execution_timestamp" type="timestamp without time zone" remarks="in UTC">
+                <constraints nullable="false" />
+            </column>
+
+            <column name="cleared" type="boolean" />
+            <column name="indication_of_collateralization" type="VARCHAR(32)">
+                <!--
+                Seems an enum value that can be normalized
+                'PC', 'UC', 'OC', 'FC', nan, 'Uncollateralized', 'Partially', 'OneWay', 'Fully'
```
Comment (du210) [inline enterprai-liquibase/dtcc-db-changelog/dtcc-changelog-1.0.xml:98]: Yep.

File: enterprai-dtcc-import/enterprai/enterprai-dtcc-csv-import/csv_importer.py [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+      "rounded_notional_amount_1_exceeded": sqlalchemy.types.Boolean,
+      "rounded_notional_amount_2_exceeded": sqlalchemy.types.Boolean,
+      "indication_of_end_user_exception": sqlalchemy.types.Boolean
+      # we should provide all the column mappings
+    }
+
+
+
+if __name__ == '__main__':
+  test_engine = sqlalchemy.create_engine("postgres://postgres:password@localhost:5433/postgres", echo=True)
```
Comment (du210) [inline enterprai-dtcc-import/enterprai/enterprai-dtcc-csv-import/csv_importer.py:113]: For the future: we should build a utility to fetch the credentials from AWS secrets manager.

File: enterprai-dtcc-import/enterprai/enterprai-dtcc-csv-import/csv_importer.py [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+  def run_import(self, path) -> None:
+    self.logger.info("Starting a new import for %s", path)
+    csv_reader = pd.read_csv(path, chunksize=self.chunk_size)
+
+    for data_frame_chunk in csv_reader:
+      self._import_data_frame(data_frame_chunk)
+
+  @abstractmethod
+  def get_column_types(self):
+    return {}
```
[the remark this answers is not in the document]
Comment (mi6091) [reply enterprai-dtcc-import/enterprai/enterprai-dtcc-csv-import/csv_importer.py:49]: Atm we are just providing the type overrides.
Having no overrides is valid too. We happen to always override it but a default implementation of an empty dictionary is also fine.

File: enterprai-dtcc-import/enterprai/enterprai-dtcc-csv-import/csv_importer.py [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+"""
+
+class CsvImporter(object):
+  def __init__(self, sql_engine, chunk_size=1000, insert_batch_size=100):
+    self.chunk_size = chunk_size
+    self.sql_engine = sql_engine
+    self.insert_batch_size = insert_batch_size
+    self.logger = logging.getLogger(self.__class__.__name__)
+
+  def _clean_data_frame(self, data_frame: pd.DataFrame) -> pd.DataFrame:
```
[the remark this answers is not in the document]
Comment (mi6091) [reply enterprai-dtcc-import/enterprai/enterprai-dtcc-csv-import/csv_importer.py:19]: Yes, it's a good idea. Probably we can use composition to enable different cleaners as most of the columns are the same, so the cleaning for them should be similar.
I am just inlining the the few steps that we have for now.

**Pull request merged**